src/utils/chartJs/CustomTooltip.jsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { withStyles } from '@material-ui/core/styles';
+
+const styles = {
+  tooltip: {
+    position: 'absolute',
+    padding: '6px',
+    background: 'rgba(0, 0, 0, .8)',
+    color: 'white',
+    borderRadius: '4px',
+    pointerEvents: 'none',
+  },
+  tooltipKey: {
+    display: 'inline-block',
+    width: '10px',
+    height: '10px',
+    marginRight: '10px',
+  },
+};
+
+class CustomTooltip extends React.Component {
+  constructor(props) {
+    super(props);
+    this.tooltip = React.createRef();
```

Comment:
I do not understand why you need this. I would assume this component will render to whatever size the `render()` produces.

Reply:
I suppose you are asking about `this.tooltip = React.createRef()`. We need a way to measure the width and height of the CustomTooltip in order to locate it in the 3x2 positions around the data point (e.g. if you want to move the tooltip over the data point, first get the position of the chart, then add position of the data point, then subtract height of tooltip). 
And because we are styling such location inside the same component, I only thought of React Refs to target and access such DOM information.

Reply:
ok, this is good enough for now.  I would imagine that we can mess with the CSS absolute positioning (relative to the point) instead of doing actual math (like is done now).